src/Go/boardAnalysis/goAI.ts

```diff
@@ -63,6 +72,18 @@ export function makeAIMove(boardState: BoardState): Promise<Play> {
   return Go.nextTurn;
 }
 
+/**
+ * Resolves the current turn.
+ * This is used for players manually playing against their script on the no-ai board.
+ */
+export function resolveCurrentTurn() {
+  if (currentTurnResolver) {
+    currentTurnResolver(Promise.resolve(getPreviousMoveDetails()));
```

Comment:
Optional: Rather than having things spread out like this, to me it would make more sense if `currentTurnResolver` were a function that takes no arguments. You'd just construct a fresh closure at the point where you're currently making this function, and all the logic would be apparent there.

Reply:
I don't think I see how that would work. It needs to be a new promise for each move, then resolve with the player's last move details as white, once they finish moving

Reply:
Hmm. I haven't been around much recently, but I should try and catch you on discord. The things I thought would be clear, clearly aren't. XD

Reply:
I think I understand what you mean now. I have changed currentTurnResolver() to not take any arguments, and instead call the resolver on Go.nextTurn and pass it the previous move's details via a closure built when the No Ai promise is constructed. I've added some comments to try and make that more clear as well

Reply:
Yup, you figured out what I was trying to get at! 👍